src/lib.rs

```diff
@@ -316,37 +339,57 @@ pub trait Backend {
     async fn exec_atomic_write(&self, op: AtomicWriteOperation<'_>) -> Result<bool>;
 }
 
-pub struct GetResult {
-    rx: mpsc::Receiver<Value>,
+pub struct GetResult(Arc<GetInner>);
+
+struct GetInner {
+    key: ExplicitKey<'static>,
+    value: std::sync::Mutex<Option<Value>>,
+}
+
+impl GetInner {
+    fn put(&self, value: Value) {
+        let mut l = self.value.lock().expect("GetInner should not be poisoned");
+        assert!(l.is_none(), "GetInner should not be double-put");
```

Comment:
thoughts on using a `OnceCell` as it closely matches the target behavior?

Reply:
Good call, thanks! I switched to `OnceLock` (not `OnceCell` because it's in an `Arc` with two references at the time). I considered this before but incorrectly thought you couldn't `take` from a `OnceLock`. But you can, once you have an exclusive reference to it, and I later made that happen anyway via `Arc::into_inner`.